src/crewai/tools/tool_usage.py

```diff
@@ -166,6 +167,26 @@ def _use(
                 if self.task:
                     self.task.increment_tools_errors()
 
+        if self.agent:
+            event_data = {
+                "agent_key": self.agent.key if self.agent else None,
+                "agent_role": self.agent.role if self.agent else None,
```

Comment:
you're checking already if agent exists

```suggestion
                "agent_key": self.agent.key,
                "agent_role": self.agent.role,
```

Reply:
Indeed, nice catch